lib/checkout_sdk/payments/hosted/hosted_payments_session.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module CheckoutSdk
+  module Payments
+    # @!attribute amount
+    #   @return [Integer]
+    # @!attribute currency
+    #   @return [String] {CheckoutSdk::Common::Currency}
+    # @!attribute payment_type
+    #   @return [String] {PaymentType}
+    # @!attribute payment_ip
+    #   @return [String]
+    # @!attribute billing_descriptor
+    #   @return [BillingDescriptor]
+    # @!attribute reference
+    #   @return [String]
+    # @!attribute description
+    #   @return [String]
+    # @!attribute customer
+    #   @return [CheckoutSdk::Common::CustomerRequest]
+    # @!attribute shipping
+    #   @return [ShippingDetails]
+    # @!attribute billing
+    #   @return [BillingInformation]
+    # @!attribute recipient
+    #   @return [PaymentRecipient]
+    # @!attribute processing
+    #   @return [ProcessingSettings]
+    # @!attribute allow_payment_methods
+    #   @return [Array(CheckoutSdk::Common::PaymentSourceType)]
+    # @!attribute products
+    #   @return [Array(CheckoutSdk::Common::Product)]
+    # @!attribute risk
+    #   @return [RiskRequest]
+    # @!attribute success_url
+    #   @return [String]
+    # @!attribute cancel_url
+    #   @return [String]
+    # @!attribute failure_url
+    #   @return [String]
+    # @!attribute metadata
+    #   @return [Hash(String=>Object)]
+    # @!attribute locale
+    #   @return [String]
+    # @!attribute three_ds
+    #   @return [ThreeDSRequest]
+    # @!attribute capture
+    #   @return [TrueClass, FalseClass]
+    # @!attribute capture_on
+    #   @return [Time]
+    # @!attribute processing_channel_id
+    #   @return [String] - Not available on Previous.
+    # @!attribute marketplace
+    #   @deprecated Please use {#amount_allocations} instead
+    #   @return [Array(CheckoutSdk::Common::MarketplaceData)] - Not available on Previous.
```

Comment:
I think this is not an array based on API reference
https://api-reference.checkout.com/#operation/createAHostedPaymentsSession!path=marketplace&t=request